spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/data/AvroDataTest.java

```diff
@@ -65,7 +91,7 @@ public abstract class AvroDataTest {
           required(117, "dec_38_10", Types.DecimalType.of(38, 10)) // Spark's maximum precision
           );
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir protected Path temp;
```

Comment:
As I mentioned on the PR for 3.4, this JUnit 4 temp folder wasn't working for JUnit 5 parameterized tests. I made some tests independent of this (to keep the backport small) and ended up porting subclasses of `AvroDataTest` to JUnit 5 in a larger backport.

These test changes were the only significant deviations from the original PRs.